docs/learn/learn-xcm-pallet.md

```diff
@@ -8,14 +8,119 @@ slug: ../learn-xcm-pallet
+defines a set of extrinsics that can be utilized to build XCVM programs, either to target the local
+or external chains. `pallet-xcm`'s functionality is separated into three categories:
 
-1. Primitive, dispatchable functions to locally execute an XCM message.
+:::note
+
+Remember, all XCMs are XCVM programs that follow
+[the XCM format](https://github.com/paritytech/xcm-format). It is the job of the XCM executor is to
+handle and execute these programs.
+
+:::
+
+1. Primitive, dispatchable functions to locally execute an XCM.
 2. High-level, dispatchable functions for asset transfers.
 3. Version negotiation-specific dispatchable functions.
+
+## Primitive Extrinsics
+
+There are two primary primitive extrinsics. These extrinsics handle sending and executing XCVM
+programs as dispatchable functions within the pallet.
+
+1. `execute` - This call contains direct access to the XCM executor. It is tje job of the executor
+   to check the message and ensure that no barrier/filter will block the execution of the XCM. Once
+   it is deemed valid, the message will then be _locally_ executed, therein returning the outcome as
+   an event. This operation is executed on behalf of whichever account has signed the extrinsic.
+   It's possible for only a partial execution to occur.
+2. `send` - This call specifies where a message should be sent externally to a particular
+   destination, i.e. a parachain, smart contract, or any system which is governed by consensus. As
+   before with `execute`, the Executor checks the origin, destination, and message and is then sent
+   to the `XcmRouter`.
+
+:::info
+
+The XCM pallet needs the `XcmRouter` to send XCMs. It is used to dictate where XCMs are allowed to
+be sent, and which XCM transport protocol to use. For example, Kusama, the canary network, uses the
+`ChildParachainRouter` which only allows for Downward Message Passing from the relay to parachains
+to occur.
+
+You can read more about [XCM transport methods here](./learn-xcm-transport.md).
+
+:::
+
+## Asset Transfer Extrinsics
+
+Several extrinsics within the pallet handle asset transfer logic. They define a predetermined set of
+instructions for sending and executing XCMs. Two variants of these functions are prefixed with
+`limited_`. They have the same functionality but can specify a weight to pay for the XCM fee.
+
+Otherwise, the fee is taken as needed from the asset being transferred.
+
+1. `reserve_transfer_assets` - Transfer some assets from the local chain to the sovereign account of
+   a destination chain and forward an XCM containing a `ReserveAssetDeposited` instruction, which
+   serves as a notification.
+2. `teleport_assets` - Teleport some assets from the local chain to some destination chain.
+
+### Transfer Reserve vs. Teleport
+
+While both extrinsics deal with transferring assets, they exhibit fundamentally different behavior.
+
+- **Teleporting** an asset implies a two-step process: the asset is burned/destroyed in the origin
```

Comment:
They're not necessarily always burnt or destroyed; what is always true though is that the assets being teleported are removed from active circulation.

Reply:
I guess this is subjective to how one implements [`checks_out`](https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/traits/transact_asset.rs#L66), but I will add that detail.